packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts

```diff
@@ -46,15 +48,39 @@ export class HostedPluginProcess implements ServerPluginRunner {
     protected readonly pluginHostEnvironmentVariables: ContributionProvider<PluginHostEnvironmentVariable>;
 
     private childProcess: cp.ChildProcess | undefined;
+
+    // child processes are kept for one minute in order to reuse them in case of network disconnections
+    private static cachedCPMap: { [clientId: string]:  {cp: cp.ChildProcess, toBeKilledAfter: number}; } = {};
```

Comment:
hello, question, why static for the map and the clienID ? I'm not sure it's shared at some point

Reply:
The scope of this class instance is one per connection between server and browser.
The whole module in packages/plugin-ext/src/hosted/node/plugin-ext-hosted-backend-module.ts is `ConnectionContainerModule`.
Each reconnect is with new instance and also different browser tabs will reconnect to different instances of this class as far as I experience.

Reply:
static properties are bad because they live outside of DI context, it would be better to have a singleton instance created from the application DI container that it is shared between all connections and inject it here.

Reply:
but wait if this class is a per connection then why do you need an id, a map and a seq at all? There can be only one client per a connection, so you should take care only about this client. Each instance of `HostedPluginProcess` will take care about own client.

Reply:
As far as I can see upon connection (new or reconnect) the same client is assigned new server and the old one is disposed.

https://github.com/theia-ide/theia/blob/843c7a0923e07d0c2b7a2f0abe04618c4b69ab08/packages/plugin-ext/src/hosted/node/plugin-ext-hosted-backend-module.ts#L49-L52

Even more than that, the DI container in which this code is run is `ConnectionContainerModule`. It is recreated on connection.

That's why I was forced to use static. If there is a way to use singleton fron another DI container that is not recreated on each reconnect I would prefer that. Or is there a simpler solution for that?

Reply:
understood, let's just extract a new class which keep mapping and inject it here instead of using static properties